plugins/plugin-svelte/plugin.js

```diff
@@ -91,15 +101,46 @@ module.exports = function plugin(snowpackConfig, pluginOptions = {}) {
       'svelte-hmr/runtime/hot-api-esm.js',
       'svelte-hmr/runtime/proxy-adapter-dom.js',
     ],
+
+    /**
+     * If any files imported the given file path, mark them as changed.
+     * @private
+     */
+    _markImportersAsChanged(filePath) {
+      if (importedByMap.has(filePath)) {
+        const importedBy = importedByMap.get(filePath);
+        importedByMap.delete(filePath);
+        for (const importerFilePath of importedBy) {
+          this.markChanged(importerFilePath);
+        }
+      }
+    },
+
+    /**
+     * When a file changes, also mark it's importers as changed.
+     * svelte.preprocess returns a list of preprocess deps - https://svelte.dev/docs#svelte_preprocess
+     */
+    onChange({filePath}) {
+      this._markImportersAsChanged(filePath);
+    },
+
     async load({filePath, isHmrEnabled, isSSR}) {
+      let dependencies = [];
       let codeToCompile = await fs.promises.readFile(filePath, 'utf-8');
       // PRE-PROCESS
       if (preprocessOptions !== false) {
-        codeToCompile = (
-          await svelte.preprocess(codeToCompile, preprocessOptions, {
+        ({code: codeToCompile, dependencies} = await svelte.preprocess(
+          codeToCompile,
+          preprocessOptions,
+          {
             filename: filePath,
-          })
-        ).code;
+          },
+        ));
+      }
+
+      // in dev mode, track preprocess dependencies
+      if (isDev && dependencies) {
```

Comment:
This should check the length of the `dependencies` array. Currently this will always be true resulting in unnecessary work.